### PR TITLE
Feature/add  pages for 404, 422, 500 errors

### PR DESCRIPTION
**Pull request by Andriy1328**
Task - https://trello.com/c/s8reYkHV/10-add-base-pages-for-errors-404-422-500
Added ErrorsController and view for error pages

### CODE REVIEW

**Review by olcplif — APPROVED** [review]
lgtm. please add a description

**Review by Pakholchuk-Oleh — APPROVED** [review]
lgtm

File: app/assets/stylesheets/errors.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+.error-button .action-link-wrap {
+  margin-top: 40px;
+
+  a {
+    background: $black-color;
+    padding: 8px 25px;
+    border-radius: 4px;
+    color: #FFF;
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/errors.scss:40]: you have a variable with this color, why not use it?

**Review by MishaKlymenko — CHANGES_REQUESTED** [review]

**Review by Bilanuk — APPROVED** [review]

File: app/helpers/errors_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module ErrorsHelper
```
Comment (Bilanuk) [inline app/helpers/errors_helper.rb:1]: do we actually need this?

File: spec/helpers/errors_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
```
Comment (Bilanuk) [inline spec/helpers/errors_helper_spec.rb:1]: do we need this file?

File: spec/views/errors/internal_server_error.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (Bilanuk) [inline spec/views/errors/internal_server_error.html.erb_spec.rb:1]: do we need tests for views?

File: spec/helpers/errors_helper_spec.rb (hunk 3 above)
Comment (igorkasyanchuk) [reply spec/helpers/errors_helper_spec.rb:1]: read first how to write specs or try by your own and see if needed or not

**Comment by igorkasyanchuk** [comment]
I really suggest you to finish all previous PR's together because they are more important, otherwise you won't learn anything useful

File: spec/helpers/errors_helper_spec.rb (hunk 3 above)
Comment (Bilanuk) [reply spec/helpers/errors_helper_spec.rb:1]: app/helpers/errors_helper.rb is an empty file.

File: spec/views/errors/internal_server_error.html.erb_spec.rb (hunk 4 above)
Comment (Andriy1328) [reply spec/views/errors/internal_server_error.html.erb_spec.rb]: fixed, no we don't

File: spec/helpers/errors_helper_spec.rb (hunk 3 above)
Comment (Andriy1328) [reply spec/helpers/errors_helper_spec.rb]: fixed, no we don't need that empty file

File: spec/requests/errors_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    end
+  end
+
+  describe "GET /internal_server_error" do
+    it "returns http success" do
+      get "/errors/internal_server_error"
+      expect(response).to have_http_status(:success)
+    end
+  end
+
```
Comment (mmrshk) [inline spec/requests/errors_spec.rb:24]: remove redundant line

Comment (mmrshk) [reply spec/requests/errors_spec.rb:24]: run rubocop here, it will raise errors

File: spec/requests/errors_spec.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe "Errors", type: :request do
+  describe "GET /not_found" do
+    it "returns http success" do
+      get "/errors/not_found"
+      expect(response).to have_http_status(:success)
```
Comment (mmrshk) [inline spec/requests/errors_spec.rb:7]: status must be not_found, not success

File: spec/requests/errors_spec.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    end
+  end
+
+  describe "GET /unprocessable_entity" do
+    it "returns http success" do
+      get "/errors/unprocessable_entity"
+      expect(response).to have_http_status(:success)
```
Comment (mmrshk) [inline spec/requests/errors_spec.rb:14]: status must be unprocessable_entity, not success

File: spec/requests/errors_spec.rb (hunk 5 above)
Comment (Andriy1328) [reply spec/requests/errors_spec.rb]: removed and fixed

File: spec/requests/errors_spec.rb (hunk 6 above)
Comment (Andriy1328) [reply spec/requests/errors_spec.rb]: fixed

File: spec/requests/errors_spec.rb (hunk 7 above)
Comment (Andriy1328) [reply spec/requests/errors_spec.rb]: fixed

File: app/assets/stylesheets/errors.scss [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    margin-right: 10px;
+
+    &:hover {
+      background: $blue-color;
+      color: $white-color;
+    }
+  }
+}
+
+#Polygon-1 , #Polygon-2 , #Polygon-3 , #Polygon-4, #Polygon-5 {
```
Comment (MishaKlymenko) [inline app/assets/stylesheets/errors.scss:54]: for future:
#Polygon-1 , 
#Polygon-2 , 
#Polygon-3 , 
#Polygon-4, 
#Polygon-5 {
...
}

**Review by MishaKlymenko — APPROVED** [review]

File: app/assets/stylesheets/errors.scss (hunk 8 above)
Comment (Andriy1328) [reply app/assets/stylesheets/errors.scss:54]: ok, thanks

**Pull request merged**